src/service/xhr-impl.js

```diff
@@ -147,6 +149,29 @@ export class Xhr {
     return this.fetch(input, setupInit(opt_init, 'text/plain'));
   }
 
+  /**
+   * A subsitute for the standard response.json(), which may optionally strip a prefix before calling JSON.parse().
+   *
+   * @param {!Response} res fetch response to convert to json.
+   * @param {string|undefined} prefix to strip away.
+   * @return {Promise<*>}
+   */
+  xssiJson(res, prefix) {
+    if (!prefix) {
+      return res.json();
+    }
+
+    return res.text().then(txt => {
+      if (!startsWith(txt, dev().assertString(prefix))) {
+        user().warn(
+          `Attempted to strip prefix "${prefix}" from xhr request, but prefix was not present.`
```

Comment:
Nit: "Failed to strip missing prefix "${prefix}" in fetch response."

Reply:
Done